pkg/flags/flags.go

```diff
@@ -44,6 +44,8 @@ func setLogLevel(logLevel string) {
 		flag.Set("stderrthreshold", "INFO")
 		flag.Set("logtostderr", "true")
 		flag.Set("v", "6") // At 7 and higher, authorization tokens get logged.
+		// pipe klog entries to logrus
+		klog.SetOutput(log.StandardLogger().Writer())
```

Comment:
FWIW i tried setting `-v=10` on the public-api container (which worked!) but didn't seem to affect the log output. i think because we're doing `flag.Set("v", "6")` after calling `flag.Parse()`, we're overriding the value.
```yaml
containers:
- args:
  - public-api
  - -prometheus-url=http://linkerd-prometheus.linkerd.svc.cluster.local:9090
  - -controller-namespace=linkerd
  - -log-level=debug
  - -v=10
```

```bash
$ kubectl -n linkerd logs -f linkerd-controller-59db74787d-w44fz public-api
time="2019-08-23T01:19:39Z" level=info msg="running version git-42ee2e2f"
time="2019-08-23T01:19:39Z" level=info msg="I0823 01:19:39.103678       1 merged_client_builder.go:122] Using in-cluster configuration"
time="2019-08-23T01:19:39Z" level=info msg="I0823 01:19:39.113756       1 round_trippers.go:438] POST https://10.96.0.1:443/apis/authorization.k8s.io/v1/selfsubjectaccessreviews 201 Created in 7 milliseconds"
time="2019-08-23T01:19:39Z" level=info msg="I0823 01:19:39.115704       1 round_trippers.go:438] GET https://10.96.0.1:443/apis/linkerd.io/v1alpha2?timeout=32s 200 OK in 0 milliseconds"
time="2019-08-23T01:19:39Z" level=info msg="I0823 01:19:39.119568       1 round_trippers.go:438] POST https://10.96.0.1:443/apis/authorization.k8s.io/v1/selfsubjectaccessreviews 201 Created in 3 milliseconds"
time="2019-08-23T01:19:39Z" level=info msg="I0823 01:19:39.120051       1 merged_client_builder.go:122] Using in-cluster configuration"
time="2019-08-23T01:19:39Z" level=info msg="I0823 01:19:39.120639       1 merged_client_builder.go:122] Using in-cluster configuration"
time="2019-08-23T01:19:39Z" level=info msg="waiting for caches to sync"
time="2019-08-23T01:19:39Z" level=info msg="I0823 01:19:39.121941       1 reflector.go:131] Starting reflector *v1.Pod (10m0s) from pkg/mod/k8s.io/client-go@v0.0.0-20181213151034-8d9ed539ba31/tools/cache/reflector.go:95"
time="2019-08-23T01:19:39Z" level=info msg="I0823 01:19:39.122504       1 reflector.go:169] Listing and watching *v1.Pod from pkg/mod/k8s.io/client-go@v0.0.0-20181213151034-8d9ed539ba31/tools/cache/reflector.go:95"
time="2019-08-23T01:19:39Z" level=info msg="I0823 01:19:39.122762       1 reflector.go:131] Starting reflector *v1beta2.ReplicaSet (10m0s) from pkg/mod/k8s.io/client-go@v0.0.0-20181213151034-8d9ed539ba31/tools/cache/reflector.go:95"
```

Reply:
So klog's level is 0 or 6 depending on whether `--controller-log-level` is `debug` or not. You think it's worth not hard-coding it to 6 if `-v` is explicitly set?

Reply:
Sorry, yeah leave the hard-coded override as-is. When I said `(which worked!)`, I meant the public-api successfully booted rather than error out with `flag provided but not defined: -foo`.